packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -523,6 +530,43 @@ export abstract class FluidDataStoreContext
 		return factory;
 	}
 
+	createChildDataStoreSync<T extends IFluidDataStoreFactory>(
+		childFactory: T,
+	): ReturnType<Exclude<T["createDataStore"], undefined>> {
+		const maybe = this.registry?.get(childFactory.type);
+
+		const isUndefined = maybe === undefined;
+		const isPromise = isPromiseLike(maybe);
+		const diffInstance = isPromise || maybe?.IFluidDataStoreFactory !== childFactory;
+
+		if (isUndefined || isPromise || diffInstance) {
```

Comment:
I wonder if you can simplify and remove all other checks other than maybe?.IFluidDataStoreFactory !== childFactory.
I think it's sufficient.

Reply:
this was mainly for unit testing to make it easier to ensure all conditions are covered. i agree it a bit verbose